File: java/java-samples/src/main/java/tech/techstay/io/FileIO.java
package tech.techstay.io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class FileIO {
  public static void main(String[] args) throws IOException {

    // reading files
    Path hosts = Paths.get("C:\\Windows\\system32\\drivers\\etc\\hosts");
    Files.readAllLines(hosts).forEach(System.out::println);

    // writing files
    Path file = Paths.get("hello.txt");
    Files.writeString(file, "hello world!");

    if (Files.exists(file)) {
      System.out.println("file created successfully");
      Files.delete(file);
    }
  }

}

File: cpp/cpp-sample/io/file_io.cpp
#include <fstream>
#include <print>

using namespace std;

int main()
{
    constexpr auto HOST_WIN   = "C:\\Windows\\System32\\drivers\\etc\\hosts"sv;
    constexpr auto HOST_LINUX = "/etc/hosts"sv;

    ifstream host_file(static_cast<string>(HOST_LINUX), fstream::in);
    for (char c; host_file.get(c);)
    {
        std::print("{}", c);
    }
    host_file.close();

    ofstream out("hello.txt", fstream::out);
    out << std::format("Hello techstay!\n");
    out.close();
}
